# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build sdl-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

if PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists guilib-build; then
  GUILIB=1
  REQUIRES="$REQUIRES guilib-build"
fi

# configure (ac-2.61; am-1.10; lt-2.2.6)
libtoolize -c -f; aclocal -I acinclude; autoconf; automake -a -c
PKG_CONFIG_PATH=/x11/lib/pkgconfig LDFLAGS=-Wl,-rpath,/x11/lib ./configure --disable-static --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/bin
install .libs/showinterfaces $DESTDIR/x11/bin
test $GUILIB && install .libs/chat $DESTDIR/x11/bin
test $GUILIB && install .libs/chatd $DESTDIR/x11/bin
install -d $DESTDIR/x11/include/SDL
install -m 644 SDL_net.h $DESTDIR/x11/include/SDL
install -d $DESTDIR/x11/lib
install .libs/libSDL_net-1.2.so.0.* $DESTDIR/x11/lib/libSDL_net-1.2.so.0
ln -sf libSDL_net-1.2.so.0 $DESTDIR/x11/lib/libSDL_net.so
install -d $DESTDIR/x11/lib/pkgconfig
install -m 644 SDL_net.pc $DESTDIR/x11/lib/pkgconfig
test $GUILIB && install -d $DESTDIR/x11/share/SDL_net
test $GUILIB && install -m 644 quit.bmp $DESTDIR/x11/share/SDL_net
test $GUILIB && install -m 644 scroll_dn.bmp $DESTDIR/x11/share/SDL_net
test $GUILIB && install -m 644 scroll_up.bmp $DESTDIR/x11/share/SDL_net

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/sdl_net-build.pc << EOF
Name: SDL_net
Version: 1.2.8
Description: Network library for SDL
Requires: $REQUIRES

devel=\\
/x11/include/SDL/SDL_net.h \\
/x11/lib/libSDL_net.so \\
/x11/lib/pkgconfig/SDL_net.pc

exec=\\
EOF
test $GUILIB && cat >> $DESTDIR/x11/share/pkgconfig/sdl_net-build.pc << EOF
/x11/bin/chat \\
/x11/bin/chatd \\
EOF
cat >> $DESTDIR/x11/share/pkgconfig/sdl_net-build.pc << EOF
/x11/bin/showinterfaces \\
/x11/lib/libSDL_net-1.2.so.0 \\
EOF
test $GUILIB && cat >> $DESTDIR/x11/share/pkgconfig/sdl_net-build.pc << EOF
/x11/share/SDL_net/quit.bmp \\
/x11/share/SDL_net/scroll_dn.bmp \\
/x11/share/SDL_net/scroll_up.bmp \\
EOF
sed -i '$ s/ \\//' $DESTDIR/x11/share/pkgconfig/sdl_net-build.pc

exit

# patch
--- SDL_net-1.2.8.orig/Makefile.am
+++ SDL_net-1.2.8/Makefile.am
@@ -1,5 +1,7 @@
 # Makefile.am for the SDL sample image loading library and viewer
 
+AUTOMAKE_OPTIONS = foreign
+
 lib_LTLIBRARIES = libSDL_net.la
 
 libSDL_netincludedir = $(includedir)/SDL
@@ -57,6 +59,8 @@
 showinterfaces_SOURCES = showinterfaces.c
 showinterfaces_LDADD = libSDL_net.la
 chat_SOURCES = chat.cpp chat.h
+chat_CPPFLAGS = -DDATADIR=\"$(pkgdatadir)\"
+chat_CXXFLAGS = -I$(includedir)/GUI
 chat_LDADD = libSDL_net.la -lGUI
 chatd_SOURCES = chatd.c
 chatd_LDADD = libSDL_net.la
--- SDL_net-1.2.8.orig/chat.cpp
+++ SDL_net-1.2.8/chat.cpp
@@ -18,9 +18,6 @@
      misrepresented as being the original software.
   3. This notice may not be removed or altered from any source distribution.
 */
-    Pleasanton, CA 94588 (USA)
-    slouken@devolution.com
-*/
 
 /* Note that this isn't necessarily the way to run a chat system.
    This is designed to excercise the network code more than be really
@@ -32,13 +29,10 @@
 #include <string.h>
 
 #include "SDL_net.h"
-#ifdef macintosh
+
 #include "GUI.h"
 #include "GUI_widgets.h"
-#else
-#include <GUI/GUI.h>
-#include <GUI/GUI_widgets.h>
-#endif
+
 #include "chat.h"
 
 
@@ -62,7 +56,7 @@
 	NUM_IMAGES
 };
 char *image_files[NUM_IMAGES] = {
-	"quit.bmp", "scroll_up.bmp", "scroll_dn.bmp"
+	DATADIR"/quit.bmp", DATADIR"/scroll_up.bmp", DATADIR"/scroll_dn.bmp"
 };
 SDL_Surface *images[NUM_IMAGES];
 
--- SDL_net-1.2.8.orig/configure.in
+++ SDL_net-1.2.8/configure.in
@@ -1,5 +1,6 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(README)
+AC_CONFIG_MACRO_DIR(acinclude)
 
 dnl Set various version strings - taken gratefully from the GTk sources
 
# source
https://www.libsdl.org/projects/SDL_net/release/SDL_net-1.2.8.tar.gz
